Move the configuration directory to ~/.config/sgpt

sywesk opened the pull request.

malonaz commented on README.md:
````diff
@@ -60,15 +60,15 @@ sgpt embed --force
 ```
 
 ## Configuration
-SGPT uses a configuration file to manage various settings. The configuration file is a JSON that includes information such as the OpenAI API Key, the API host, request timeout, the default model, and the chat directory. This resides in a default directory `~/.sgpt/internal/config.json`.
````
J ai fait une erreur ici. Y'a pas d"internal" dsns le config path.

malonaz reviewed (approved):
All good. Let's take away the migration stuff for now. I don't think we have any actual users 🥺

sywesk committed:
[README.md] - docs: update path for configuration and chat directory

 - The default path for configuration and chat directory has been updated from ~/.sgpt/ to ~/.config/sgpt/

[internal] - refactor: update default config paths in configuration file

 - Default paths for chat and embed directories in the configuration file have been updated to new setup

[chat] - fix: correct indentation in chat.go

 - Fixed improper indentation in chat.go for better code readability

[cmd] - refactor: update config file path in main.go

 - The default config file path in main.go has been updated to align with the new setup
